### PR TITLE
Fix TileEntity references for 1.13+

### PR DESCRIPTION
**Pull request by Cadiboo**
Fixes references and corrects a couple typos.
`IBlockstate` -> `Blockstate`
`NBTTagCompound` -> `CompoundNBT`
`IBlockstate` -> `Blockstate`
`ITickable` -> `ITickableTileEntity`
`SPacketUpdateTileEntity` -> `SUpdateTileEntityPacket`

### CODE REVIEW

File: docs/tileentities/tileentity.md [hunk 1]
```diff
@@ -26,9 +26,9 @@ public static void registerTE(RegistryEvent.Register<TileEntityType<?>> evt) {
 
 To attach your new `TileEntity` to a `Block` you need to override 2 (two) methods within the Block class.
 ```JAVA
-IForgeBlock#hasTileEntity(IBlockstate state)
+IForgeBlock#hasTileEntity(Blockstate state)
```
Comment (tterrag1098) [inline docs/tileentities/tileentity.md]: It's `BlockState`

File: docs/tileentities/tileentity.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ To register it, listen for the appropriate registry event and create a `TileEnti
 ```Java
 @SubscribeEvent
 public static void registerTE(RegistryEvent.Register<TileEntityType<?>> evt) {
-  TileEntityType<?> type = TileEntityType.Builder.create(<factory>).build(null)
+  TileEntityType<?> type = TileEntityType.Builder.create(<factory>, <block...>).build(null);
```
Comment (tterrag1098) [inline docs/tileentities/tileentity.md]: Use valid syntax here, or at least a comment

File: docs/tileentities/tileentity.md [hunk 3]
```diff
@@ -92,30 +92,30 @@ This method is a bit more complicated, but again you just need to override 2 met
...
+* An `CompoundNBT` which should contain your data.
 
 Now, to send the packet, an update notification must be given on the server.
 ```JAVA
-World#notifyBlockUpdate(BlockPos pos, IBlockState oldState, IBlockState newState, int flags)
+World#notifyBlockUpdate(BlockPos pos, BlockState oldState, BlockState newState, int flags)
 ```
 The `pos` should be your TileEntitiy's position. For `oldState` and `newState` you can pass the current BlockState at that position.
-The `flags`are a bitmask and should contain 2, which will sync the changes to the client.
+The `flags` are a bitmask and should contain 2, which will sync the changes to the client.
```
Comment (tterrag1098) [inline docs/tileentities/tileentity.md]: Perhaps mention `Constants.BlockFlags`

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
Please rebase this to 1.14.x

File: docs/tileentities/tileentity.md [hunk 4]
```diff
@@ -16,17 +16,18 @@ To register it, listen for the appropriate registry event and create a `TileEnti
 ```Java
 @SubscribeEvent
 public static void registerTE(RegistryEvent.Register<TileEntityType<?>> evt) {
-  TileEntityType<?> type = TileEntityType.Builder.create(<factory>, <block...>).build(null);
+  TileEntityType<?> type = TileEntityType.Builder.create(factory, validBlocks).build(null);
   type.setRegistryName("mymod", "myte");
   evt.getRegistry().register(type);
 }
 ```
+In this example, `factory` is a function that creates a new instance of your TileEntity. A method reference or a lamda is commonly used. The variable `validBlocks` is one or more blocks (`TileEntityType.Builder#create` is varargs) that the tile entity can exist for.
```
Comment (tterrag1098) [inline docs/tileentities/tileentity.md]: lambda*

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

**Comment by tterrag1098** [comment]
Looks good, but it still needs to be rebased.

**Comment by Cadiboo** [comment]
Rebased to 1.14.x.

**Review by tterrag1098 — APPROVED** [review]

**Pull request merged**